src/modules/upnp/XmlFunctions.cpp

```diff
@@ -112,7 +112,11 @@ QString XmlFunctions::getNodeValue(const QDomNode & rootNode, const QString & pa
 QString XmlFunctions::getSource(const QDomNode & node, int indent)
 {
 	QString source;
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
 	QTextStream textStream(&source, QIODevice::WriteOnly);
+#else
+	QTextStream textStream(&source, QIODeviceBase::WriteOnly);
```

Comment:
QIODevice::WriteOnly should still work in Qt6, because QIODevice inherits from QIODeviceBase

Reply:
```
src/modules/upnp/XmlFunctions.cpp:115:52: error: incomplete type ‘QIODevice’ used in nested name specifier
  115 |         QTextStream textStream(&source, QIODevice::WriteOnly);
```
You are right, this could be fixed with just an `#include <QIODevice>`

Reply:
Done in https://github.com/kvirc/KVIrc/pull/2557/commits/4e922196f409363439c65facfaef3605671716db